### PR TITLE
Fix: #1142 #1143 eslint configuration for stencil app generator fails

### LINKED ISSUE #1142: Stencil: `.eslintrc.json` cannot be found while generating `stencil` app
**Describe the bug**
obraz

**To Reproduce**

1. `npx create-nx-workspace@latest --preset=apps`
2. `npm i @nxext/stencil -D`
3. `npx nx generate @nxext/stencil:app apps/stencil-app --verbose`

**Expected behavior**
created `stencil` app

**Additional context**

### LINKED ISSUE #1143: [Question] Stencil: pinned `@stencil/core` dependency
**Describe the bug**

Hi, I wanted to check if there’s a particular reason why the `@stencil/core` dependency has a fixed version.

CleanShot 2024-10-15 at 12 45 41

My `@nx` workspace is using the latest `@stencil/core` for our stencil components library, but `@nxext/stencil` v.20.0.1 now installs another version in the node_modules, meaning that our workspace has two versions.

CleanShot 2024-10-15 at 12 51 19

Our packages are using the right version, but I don't believe that fixing `@stencil/core` to a specific version is the best approach. I might be overlooking something.

**To Reproduce**

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by pawel-twardziak**
Closes #1142 
Closes #1143

### CODE REVIEW

**Comment by pawel-twardziak** [comment]
@DominikPieper @edbzn sorry for mixing and matching in this PR - it contains a feature for common and a fix for stencil.

**Comment by pawel-twardziak** [comment]
Damn, need to split it up 😭

**Comment by pawel-twardziak** [comment]
this is on hold waiting for #1145 to be merged and released

**Comment by edbzn** [comment]
Seems like you need to rebase to get the CI passing @pawel-twardziak.

File: packages/stencil/package.json [hunk 1]
```diff
@@ -26,9 +26,9 @@
   "builders": "./executors.json",
   "executors": "./executors.json",
   "dependencies": {
-    "@nxext/common": "19.0.0 || ^20.0.0",
+    "@nxext/common": "19.1.0 || ^20.0.0",
```
Comment (edbzn) [inline packages/stencil/package.json:29]: We have to be more strict here and only accept the @nxext/common `^20.0.0`.

File: packages/stencil/src/utils/lint.ts [hunk 2]
```diff
@@ -34,6 +46,96 @@ export function getEsLintPluginBaseName<
   throw Error(`[stencil] unsupported eslint plugin name: ${packageName}`);
 }
 
+export const augmentStencilEslintFlatConfig = (
```
Comment (edbzn) [inline packages/stencil/src/utils/lint.ts:49]: I think we need a unit test to validate the final shape of the eslint config file.

**Comment by pawel-twardziak** [comment]
> Seems like you need to rebase to get the CI passing @pawel-twardziak.

yes, it was waiting for one of the previous PRs. Now rebased, still some test to fix :)

File: packages/stencil/src/utils/lint.ts (hunk 2 above)
Comment (pawel-twardziak) [reply packages/stencil/src/utils/lint.ts:60]: Done

**Review by pawel-twardziak — COMMENTED** [review]
ready to CR

**Comment by pawel-twardziak** [comment]
@DominikPieper @edbzn ready to CR and a potential merge :)

**Review by edbzn — APPROVED** [review]

**Comment by edbzn** [comment]
Nice work @pawel-twardziak!

**Pull request merged**